src/main/java/core/basesyntax/impl/StorageImpl.java

```
@@ -1,19 +1,53 @@
 package core.basesyntax.impl;
 
 import core.basesyntax.Storage;
+import java.util.Objects;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+
+    private static final int MAX_SIZE = 10;
+    private int size = 0;
+
+    private final V[] storageContents = (V[]) new Object [MAX_SIZE];
+    private final K[] storageNames = (K[]) new Object [MAX_SIZE];
+
     @Override
     public void put(K key, V value) {
+
+        for (int i = 0; i < size; i++) {
+            if (Objects.equals(key, storageNames[i])) {
+                storageContents[i] = value;
+                size--;
+            }
+        }
+
+        try {
+            storageNames[size] = key;
+            storageContents[size] = value;
+            size++;
+        } catch (Exception e) {
+            if (size >= MAX_SIZE) {
+                throw new StorageSizeException("Can't put value to storage, "
+                        + "because storage is full");
+            }
+        }
```

Why using try/catch on code that won't throw any exception.

I think you should rethink your strategy:
You've just done iterating over the storage for indexes <0,size)
If you managed to insert object on existing key during iteration, you can exit method immediately from the loop with `return`. Size is not changes since you're overwriting the existing value.
But what, if you won't find expected object with loop?
My guess is: you must check whether the max size is reached. If it's exceeded, you should throw an exception. And if not? Just put the object as a next element and increase the size.